flytepropeller/pkg/controller/nodes/executor.go

```diff
@@ -201,6 +201,16 @@ func (c *recursiveNodeExecutor) RecursiveNodeHandler(ctx context.Context, execCo
 		// 6. Thus we can get rid of SetInputs for StartNode as well
 		logger.Debugf(currentNodeCtx, "Handling node Status [%v]", nodeStatus.GetPhase().String())
 
+		// if it's configured to retry with delay, we should sleep before retrying
+		if nodeStatus.GetPhase() == v1alpha1.NodePhaseRetryableFailure {
+			if currentNode.GetRetryStrategy().RetryDelay != nil {
+				sleepDuration := time.Until(nodeStatus.GetTaskNodeStatus().GetLastPhaseUpdatedAt().Add(currentNode.GetRetryStrategy().RetryDelay.Duration))
+				if sleepDuration > 0 {
+					logger.Infof(currentNodeCtx, "Sleeping for [%v] before retrying", sleepDuration)
+					time.Sleep(sleepDuration)
```

Comment:
oooh we do not want to add sleep to the event loop. We just need to return and not actually do a retry till the time has elapsed.

Reply:
ah ok, I thought this was part of a green thread.

If we just return and the delay is configured to be let's say 1 hour. Will propeller receive the same event in a loop for an hour that will overload it?

Reply:
it wont overload, it wont check it for next refresh interval. Another way to do this is to implement a timer.wheel that can check for active timers and fire the necessary workflow event. We can do that on a separate goroutine